scripts/cpplint.py

```diff
@@ -5630,13 +5639,29 @@ def CheckForIncludeWhatYouUse(filename, clean_lines, include_state, error,
   if IsSourceFilename(filename) and not header_found:
     return
 
+  # Keep track of which headers have been reported already
+  reported = set()
+
   # All the lines have been processed, report the errors found.
-  for required_header_unstripped in required:
-    template = required[required_header_unstripped][1]
-    if required_header_unstripped.strip('<>"') not in include_dict:
-      error(filename, required[required_header_unstripped][0],
+  for template in required:
+    line_and_headers = required[template]
+    headers = line_and_headers[1]
+    found = False
+    for required_header_unstripped in headers:
+      if required_header_unstripped in reported:
+        found = True
+        break
+
+      if required_header_unstripped.strip('<>"') in include_dict:
+        found = True
+        break
+
+    if not found:
+      preferred_header = headers[0]
+      reported.add(preferred_header)
+      error(filename, line_and_headers[0],
             'build/include_what_you_use', 4,
-            'Add #include ' + required_header_unstripped + ' for ' + template)
+            'Add #include ' + preferred_header + ' for ' + template)
```

Comment:
Very optional: ideally, I'd output the alternatives as well, not only the `preferred_header`. Feel free to ignore if it's annoying to implement.

Reply:
Now outputs:

Firestore/core/test/firebase/firestore/testutil/xcgmock.h:103:  Add #include <iosfwd> for ostream (or <ostream>, <iostream>)  [build/include_what_you_use] [4]

Reply:
Thanks!